### PR TITLE
refactor(android): Clean up tunnel fd establishment

### PR DESCRIPTION
**Pull request by jamilbk**
Cleans up the VpnService establishment logic a little

### CODE REVIEW

File: kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt [hunk 1]
```diff
@@ -127,9 +126,8 @@ class TunnelService : VpnService() {
 
                 val route = Cidr(addr, prefix)
                 tunnelRoutes.add(route)
-                val fd = buildVpnService().establish()?.detachFd() ?: -1
-                protect(fd)
```
Comment (jamilbk) [inline kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt:131]: I don't believe this protect was doing anything; the Android docs say this should be used explicitly on a socket: https://developer.android.com/reference/android/net/VpnService#protect(java.net.Socket)

File: kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt [hunk 2]
```diff
@@ -199,15 +195,6 @@ class TunnelService : VpnService() {
         return START_STICKY
     }
 
-    // Happens when a user removes the VPN configuration in the System settings
```
Comment (thomaseizinger) [inline kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt:202]: Was this comment not true?

Comment (jamilbk) [reply kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt:202]: Hmm let me test this. The docs aren't clear on when this callback is invoked. It wasn't occurring in the typical tunnel lifecycle

Comment (jamilbk) [reply kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt:202]: Yeah, it doesn't seem to ever be called both when disconnecting from the App UI or by forgetting the VPN in system settings.

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**